### PR TITLE
Modernised Find and Replace UI 

### PR DESCRIPTION
**Pull request by jncn**
Modernised the Find and Replace UI, and some additional clean up:
- The Find and Replace UI has now a more space efficient lay out, I think without compromises
- Fixes a bug where the Find All window would clip text until resizing its window
- Smaller lay out improvements for the Find All window
- Removed HUD buttons style from DocumentProxy
- Removed old assets (even some nice Pather-style buttons 💔)

And of course some other commits sneaked in again, not sure why this is happening.

### CODE REVIEW

**Comment by monkeydom** [comment]
Pasted_Image_12_12_20__19_18


Addding a after and before comparison image. Overall I do think this is a positive change, but I'll need some getting used to it. The gear dropdown is a little bit cryptic, and when regex is off the actual gear does not align nicely either. `(.*)` doesn't say too much regex to me, and bracing evertyhing just makes `\1` equal `\0` so hmmm. 

So to gather some open issues:
* the inline result text overlaps and looks broken if the regex is actually filling the line
* the gear is not aligned properly
* the regex mode of the gear is not expressive enough

**Comment by monkeydom** [comment]
Pasted_Image_12_12_20__19_38
Example of a replace all result text overwriting the long regex.

**Comment by monkeydom** [comment]
Pasted_Image_12_12_20__19_43
German localizations seems to also show "1 replaced" - might be bug existing b ut should be fixed either wy.

File: SubEthaEdit-Mac/Base.lproj/Localizable.strings [hunk 1]
```diff
@@ -23,13 +23,13 @@
...
 
 /* Entries Found in FindAll Panel */
-"%d found." = "%d found.";
+"%d matches" = "%d matches";
 
 /* Pending Users Display in Menu Title Bar */
 "%d pending" = "%d pending";
 
 /* Number of replaced strings */
-"%d replaced." = "%d replaced.";
```
Comment (monkeydom) [inline SubEthaEdit-Mac/Base.lproj/Localizable.strings:32]: here is the localization culprit. if you change the keys you need to touch the german loclziation too

File: SubEthaEdit-Mac/Source/SEEFindAndReplaceState.m [hunk 2]
```diff
@@ -276,11 +276,11 @@ - (void)takeValuesFromDictionaryRepresentation:(NSDictionary *)aDictionaryRepres
 
 - (NSString *)menuTitleDescription {
 	NSMutableArray *components = [NSMutableArray new];
-	[components addObject:[NSString stringWithFormat:@"›%@‹", self.findString]];
```
Comment (monkeydom) [inline SubEthaEdit-Mac/Source/SEEFindAndReplaceState.m:279]: Don't know f I see the removal of the angle braces an improvement.

**Comment by jncn** [comment]
Alright! Thanks for the review. I'll do a proposal for the dropdown and fix the label.

File: SubEthaEdit-Mac/Source/SEEFindAndReplaceState.m (hunk 2 above)
Comment (jncn) [reply SubEthaEdit-Mac/Source/SEEFindAndReplaceState.m:279]: Fair enough! It confused me when I first saw it in that menu, I thought it was a glitch. I've never seen these in use before. 

I looked around a bit and found out other apps aren't using any braces in their recents. Which doesn't mean SEE should do the same, of course 😉

File: SubEthaEdit-Mac/Base.lproj/Localizable.strings (hunk 1 above)
Comment (jncn) [reply SubEthaEdit-Mac/Base.lproj/Localizable.strings:32]: Missed that one, good catch!

**Review by monkeydom — CHANGES_REQUESTED** [review]
Requesting some changes as in the discussion. But looks good in general.

File: SubEthaEdit-Mac/Source/SEEFindAndReplaceState.m (hunk 2 above)
Comment (monkeydom) [reply SubEthaEdit-Mac/Source/SEEFindAndReplaceState.m:279]: I do agree all of this is quite cryptic though. If you like it clearly better this way I might get convinced as less often is more.

File: SubEthaEdit-Mac/Base.lproj/Localizable.strings (hunk 1 above)
Comment (monkeydom) [reply SubEthaEdit-Mac/Base.lproj/Localizable.strings:32]: maybe it isn't? seems to be fixed in the german localization. maybe the problem is somewhere else.

**Comment by jncn** [comment]
I played around with the lay out a bit again, I think we're nearly there:
- Fixed the localisation issue
- Fixed the overlapping status text 
- Moved the options button back to the left
- Changed the RegEx icon to `*`

One open question is backwards compatibility. I find myself writing more and more checks for macOS 10.14+ (because quite a few UI-API's changed with the introduction of Dark Mode). Would it be a good idea to drop 10.13 and older for 5.2?
Because of this, I haven't tackled the issue of tinting the button when a search with a scope is performed (the shadow didn't look good).

Feedback welcome! 

Here are already some screenshots:

Screenshot 2020-12-13 at 01 03 39

Screenshot 2020-12-13 at 00 46 46

**Comment by monkeydom** [comment]
I'm open to requiring 10.14 aka mojave. It seems like a solid progression. It is also easier for me to actually make sure it works, as my 10.13 machine is probably not for much longer anyways.

And if it turns out that we do have a good case for macOS 11 only because of major style questions, I'm open to that as well, as we do have a solid release for the lower versions and I do think the macOS 11 style really begs for bigger changes that probably feel a little alien to 10.15/10.14 anyways.

**Review by monkeydom — COMMENTED** [review]
Looks good to me. Need to do the manual diff, build an test dance before actually approving.

**Review by monkeydom — APPROVED** [review]
Looks very good to me, thanks!

**Pull request merged**